engine/bsql_engine/io/io.pyx

```diff
@@ -399,10 +399,14 @@ cpdef runQueryCaller(int masterIndex,  tcpMetadata,  tables,  vector[int] fileTy
     for i in range(names.size()):
         decoded_names.append(names[i].decode('utf-8'))
 
-    df = cudf.DataFrame(CudfXxTable.from_unique_ptr(blaz_move(dereference(resultSet).cudfTable), decoded_names)._data)
-    
-    return df
-
+    if is_single_node: # the engine returns a concatenated dataframe
```

Comment:
@felipeblazing please check this part for single_node usage

Reply:
nope it can return multiples ones too like a normal query